remote-api/src/main/java/org/alfresco/rest/api/impl/SizeDetailsImpl.java

```diff
@@ -0,0 +1,212 @@
+    private static final String FOLDER = "folder";
+    private Nodes nodes;
+    private NodeService nodeService;
+    private PermissionService permissionService;
+    private ActionService actionService;
+    private SimpleCache<Serializable,Map<String, Object>> simpleCache;
+    private int defaultItems;
+
+    public void setNodes(Nodes nodes)
+    {
+        this.nodes = nodes;
+    }
+
+    public void setPermissionService(PermissionService permissionService)
+    {
+        this.permissionService = permissionService;
+    }
+
+    public void setNodeService(NodeService nodeService)
+    {
+        this.nodeService = nodeService;
+    }
+
+    public void setActionService(ActionService actionService)
+    {
+        this.actionService = actionService;
+    }
+
+    public void setSimpleCache(SimpleCache<Serializable, Map<String, Object>> simpleCache)
+    {
+        this.simpleCache = simpleCache;
+    }
+
+    public void setDefaultItems(int defaultItems)
+    {
+        this.defaultItems = defaultItems;
+    }
+
+    /**
+     * generateNodeSizeDetailsRequest : providing HTTP STATUS 202 with jobId.
+     */
+    @Override
+    public NodeSizeDetails generateNodeSizeDetailsRequest(String nodeId)
+    {
+        NodeRef nodeRef = nodes.validateNode(nodeId);
+        validateType(nodeRef);
+        String actionId;
+        if(simpleCache.get(nodeId) == null)
+        {
+            actionId = executeAction(nodeRef, defaultItems, simpleCache);
+        } else
+        {
+            Map<String, Object> result = simpleCache.get(nodeRef.getId());
+            actionId = (String)result.get(ACTIONID);
+        }
+        return new NodeSizeDetails(actionId);
+    }
+
+    /**
+     * getNodeSizeDetails : providing HTTP STATUS 200 with NodeSizeDetails data from cache.
+     */
+    @Override
+    public NodeSizeDetails getNodeSizeDetails(final String nodeId, final String jobId)
+    {
+        NodeRef nodeRef = nodes.validateNode(nodeId);
+        validateType(nodeRef);
+
+        if(simpleCache.get(nodeId) == null)
+        {
+            return new NodeSizeDetails(nodeId, NOT_INITIATED.name());
+        }
+
+        LOG.debug("Executing executorResultToSizeDetail  method");
+        return executorResultToSizeDetail(simpleCache.get(nodeId), nodeId, jobId);
+    }
+
+    /**
+     * Executing Action Asynchronously.
+     */
+    private String executeAction(NodeRef nodeRef, int defaultItems, SimpleCache<Serializable, Map<String, Object>> simpleCache)
+    {
+        Map<String, Object > currentStatus = new HashMap<>();
+        currentStatus.put(STATUS,IN_PROGRESS.name());
+        Action folderSizeAction = actionService.createAction(NodeSizeDetailActionExecutor.NAME);
+        currentStatus.put(ACTIONID,folderSizeAction.getId());
+        folderSizeAction.setTrackStatus(true);
+        folderSizeAction.setExecuteAsynchronously(true);
+        folderSizeAction.setParameterValue(NodeSizeDetailActionExecutor.DEFAULT_SIZE, defaultItems);
```

Comment:
Previously, we were utilizing the POST endpoint parameter, but as you said, we were supposed to retrieve from the value configured in the property file.
This value has been used for a large number of folders so that we can aggregate the sum in many units.